app.js

```diff
@@ -8,28 +8,127 @@ if (NODE_ENV === 'development') app.use(require('morgan')('dev'))
+      "price": "1.00"
+    },
+    {
+        "id": "SH_kBYDBM",
+        "name": "mango",
+        "price": "2.00"
+    },
+    {
+        "id": "noXMXa0bK",
+        "name": "banana",
+        "price": "0.50"
+    }
+  ],
+  vegetables: [
+    {
+      "id": "gizBnTGnI",
+      "name": "tomato",
+      "price": "1.00"
+    },
+    {
+        "id": "B-1DtOWS6",
+        "name": "potato",
+        "price": "2.00"
+    },
+    {
+        "id": "mMKcqFOE8",
+        "name": "green pepper",
+        "price": "1.20"
+    },
+    {
+        "id": "kOfsVqi6d",
+        "name": "red pepper",
+        "price": "1.50"
+    }
+  ]
 }
 
 app.get('/vegetables', (req, res, next) => {
   const { vegetables } = data
-  res.json(vegetables)
+  // Capture the query in a varaiable
+  const name = (req.query.name);
+
+  console.log(name);
+  // Check if any query param was sent
+  if (name)
+  {
+    console.log("query present");
+    // Check if the requested vegetable matches any of the ones present in the array, while checking remove the quotes
+    const results = vegetables.filter(veggie => veggie.name.includes(name.replace(/"/g,"")));
+    console.log(results)
+    res.json(results);
+  }
+  // If no query params were sent then return the entire vegetable array
+  else
+  {
+    console.log("no query present")
+    res.json(vegetables)
+  }
 })
 
-app.get('/vegetables/:id', (req, res, next) => {
+app.delete('/vegetables/:id', (req, res, next) => {
   const { vegetables } = data
   const { id } = req.params
+  // Check if there is any vegetable that matches the id
   const vegetable = vegetables.find(veggie => veggie.id === id)
 
-  if (!vegetable) {
+  // If we do not find a veggie that matches then return not found along with status 404
+  if (!vegetable) 
+  {
     const message = `Could not find vegetable with ID of ${id}`
     next({ status: 404, message })
   }
+  // If we find the veggie that matches, then remove that veggie from the array
+  // You can check this by doing a successive get which would not have the deleted veggie as a part
+  // of its complete list
+  else
+  {
+    vegetables.splice(vegetables.indexOf(vegetable), 1)
+  }
 
+  // return the deleted veggie
   res.json(vegetable)
 })
 
+app.put('/vegetables/:id', (req, res, next) => {
+  const { vegetables } = data
+  const { id } = req.params
+  // Check if there is any vegetable that matches the id
+  const vegetable = vegetables.find(veggie => veggie.id === id)
+
+  // If we do not find a veggie that matches then return not found along with status 404
+  if (!vegetable) 
+  {
+    const message = `Could not find vegetable with ID of ${id}`
+    next({ status: 404, message })
+  }
+  // If we find the veggie with the specified Id then update the same
+  else
+  {
+    // If the request does not have both the name and price return early with a bad request message
+    if ((!req.body.name) || (!req.body.price))
+    {
+      const message = "Bad request";
+      res.json({ message })
```

Comment:
This doesn't actually change the status code, unfortunately. You could instead do:
```js
res.status(400).json({ message })
```